### PR TITLE
[Cherry-pick] Merge "raftstore & raftstore-v2:control grpc server according to slowness(#15088)" to 6.5.

### LINKED ISSUE #15086: TiKV provides APIs to restart grpc service
When a TiKV node becomes unhealthy  (such as slow IO, network partition from certain nodes),  we usually want to fail fast existing requests and do leader transfer. 
However, we may not want to kill a tikv server when it's unhealthy because restarting the tikv is too expensive and it may impact performance (due to warmup cache). 
So there should be a way to do a soft restart that will only start/stop grpc services

### PR DESCRIPTION
**Pull request by LykxSassinator**
This is a manual cherry-pick of https://github.com/tikv/tikv/pull/15088

### What is changed and how it works?

Issue Number: Close #15086

What's Changed:

```commit-message
Make raftstore & raftstore-v2 able to control the grpc service, according to the slowness.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None.
```

### CODE REVIEW

**Comment by LykxSassinator** [comment]
/cc @tonyxuqqi @overvenus 
PTAL, thx.

File: components/raftstore/src/store/worker/pd.rs [hunk 1]
```diff
@@ -986,6 +991,7 @@ where
             curr_health_status: ServingStatus::Serving,
             coprocessor_host,
             causal_ts_provider,
+            _grpc_service_manager, // TODO: be used when optimization on slowness detection is ready
```
Comment (tonyxuqqi) [inline components/raftstore/src/store/worker/pd.rs:994]: you can add it when it's being used.

File: components/server/src/server2.rs
Comment (tonyxuqqi) [inline components/server/src/server2.rs:1]: V2 is not needed in v6.5

Comment (LykxSassinator) [reply components/server/src/server2.rs:1]: yep, these related parts have been reverted.

**Review by tonyxuqqi — APPROVED** [review]
overall LGTM except the "_grpc_service_manager" should be removed from pd.rs as it's not used now.

**Review by overvenus — APPROVED** [review]

**Comment by LykxSassinator** [comment]
And we have the following testing records:
img_v2_a64ad1e1-68df-4d31-a20e-9747d76a1d1g
which proved that this feature can be GA in 6.5

**Pull request merged**